app/domain/authentication/validate_origin.rb

```diff
@@ -22,7 +22,15 @@ def call
     private
 
     def role
-      @role_cls.by_login(@username, account: @account)
+      return @role if @role
+
+      @role = @role_cls.by_login(@username, account: @account)
+      raise Errors::Authentication::Security::RoleNotFound, role_id unless @role
```

Comment:
should this just be
```suggestion
      raise Err::Security::RoleNotFound, role_id unless @role
```
since `Err ||= Errors::Authentication` above?

also, does `Security::RoleNotFound` need to be added to the comment at the top of the file?

Reply:
thanks, nice catch